plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PluginEndpointDiscovery } from '@backstage/backend-common';
+import { Entity } from '@backstage/catalog-model';
+import { IndexableDocument, DocumentCollator } from '@backstage/search-common';
+import fetch from 'cross-fetch';
+import unescape from 'lodash/unescape';
+import { Logger } from 'winston';
+
+export interface TechDocsDocument extends IndexableDocument {
+  kind: string;
+  namespace: string;
+  entityKind: string;
+  entity: string;
+  lifecycle: string;
+}
```

Comment:
It's not the case in the other collator right now, but we noticed that it's quite useful if you have the type in the frontend too.
Therefore we put it into the *-common package. Not sure what the option on this is.

Reply:
I agree that we should probably do this. I'll leave it as is for now unless we find a common ground on how to handle these types. Sharing types/models across services has so many proponents and opponents that it needs a bigger discussion (and an ADR) I feel.